hive/lib/src/backend/vm/storage_backend_vm.dart

```diff
@@ -78,7 +79,9 @@ class StorageBackendVm extends StorageBackend {
     this.registry = registry;
 
     lockRaf = await _lockFile.open(mode: FileMode.write);
-    await lockRaf.lock();
+    if ((Hive as HiveImpl).useLocks) {
```

Comment:
I did consider that, however I thought it would be better to keep the useLocks value hidden from users and only allow setting it in the constructor. The cast is a pattern I observed in several places in the codebase already, for example here: https://github.com/hivedb/hive/blob/cacfc53dc1114491eaa3d3f3842312db910c1330/hive/lib/src/box_collection/box_collection.dart#L43 and here: https://github.com/hivedb/hive/blob/814594978426509290f67638b4a85039644f1570/hive/lib/src/object/hive_list_impl.dart#L55

I can of course change it, if that is what you prefer, but Hive is not a type, it is a global variable, which you call Hive.init on and such. It is defined here https://github.com/hivedb/hive/blob/cacfc53dc1114491eaa3d3f3842312db910c1330/hive/lib/hive.dart#L66 to be a HiveImpl and can't be reassigned, since it is final. I would also assume the library would break, if that was done, since there are several casts already in the library.

Reply:
Ah sorry, my bad 😄 For a moment I thought `Hive` is a type name instead of a static value.